main.c

```diff
@@ -16,194 +19,265 @@
 
 	/*
-		1: quiet batch
-		2: verbose batch
-		3: interactive
-	*/
-	int mode = INTERACTIVE_MODE;
+	 * 1: Quiet Batch Mode
+	 * 2: Verbose Batch Mode
+	 * 3: Interactive Mode
+	 */
+    int *mode = malloc(sizeof(int));
+    *mode = INTERACTIVE_MODE;
 
-	/* Parse Command Line Args */
-	char c;
+    // By default, the color functionality is off
+    *color = false;
 
-	/* If command line contains just the file at argv[1], write it into file_name */
-	if (fileexists(argv[1])) strcpy(file_name,argv[1]);
+	// Parse Command Line Arguments
+    char c;
 
+    if (fileexists(argv[optind])) {
+        strcpy(filename, argv[optind]);
+        optind++;
+    }
 
     // Parse the initial command line and 
-	while ((c = getopt(argc,argv,"d:i:v:s:q:")) != -1) {
+	while ((c = getopt(argc, argv, "ilvcqs:d:v")) != -1) {
 		switch (c) {
-		case 'd':
-			if (!fileexists(optarg)) {  // this checks if the file actually exists
-				shell_error("Dictionary input file path invalid");
-				return EXIT_FAILURE;
-			}
-			strcpy(dict_name,optarg);
-			if (mode == 3) shell_input(optarg, "dictionary");
-			break;
-		case 'i':
-			if (!fileexists(optarg)) {
-				shell_error("Input file path invalid");
-				return EXIT_FAILURE;
-			}
-			mode = 3;
-            strcpy(file_name,optarg);
-			// if file name is not valid, print error
-			break;
-		case 'v':
-			if (!fileexists(optarg)) {
-				shell_error("Input file path invalid");
-				return EXIT_FAILURE;
-			}
-			mode = 2;
-            strcpy(file_name,optarg);
-			break;
-		case 'q':
-			if (!fileexists(optarg)) {
-				shell_error("Input file path invalid");
-				return EXIT_FAILURE;
-			}
-			mode = 1;
-			strcpy(file_name,optarg);
-			break;
-		case 's':
-            if (strstr(optarg,".txt\0") == NULL) {    // does not save to a *.txt file
-                  shell_error("Input save file path invalid");
-                  return EXIT_FAILURE;
-            }
-			strcpy(save_file,optarg);
-            if (mode == 3) shell_input(optarg,"file save destination");
-			break;
-		default:
-			shell_usage();
-            exit(0);
-		}
+            case 'd':
+                if (!fileexists(optarg)) {  // this checks if the file actually exists
+                    shell_error("Invalid dictionary file input.", color);
+                    log_fatal("Dictionary file could not be found.");
+                    return EXIT_FAILURE;
+                }
+
+                strcpy(dict, optarg);
+
+                if (*mode == INTERACTIVE_MODE) {
+                    shell_input(optarg, "dictionary", color);
+                }
+
+                break;
+
+            case 'i':
+                *mode = INTERACTIVE_MODE;
+                break;
+
+            case 'l':
+                *mode = VERBOSE_MODE;
+                break;
+
+            case 'q':
+                *mode = QUIET_MODE;
+                break;
+		
+            case 's':
+                if (strstr(optarg, ".txt\0") == NULL) {    // does not save to a *.txt file
```

Comment:
Why is the user not allowed to save to whatever kind of file they want?

Reply:
this has been deleted.